### PR TITLE
Feature/reactor 431 - new CZ-NEF and CZ-NRF fields

### PR DESCRIPTION
**Pull request by killthrush**
### Description

In order to support some new features and to provide better names to some existing fields, we've extended CZ-NEF and CZ-NRF:

new CZ-NEF
-----------------------------------------
cz_account_id - character varying(512)
cloud_account_type - text
service_type - text (indexed)
resource_type - text (indexed)
event_service_host - text (indexed)
event_flags - int

new CZ-NRF
-----------------------------------------
resource_id - character varying(512)
last_observed - numeric(15,5)
first_observed - numeric(15,5)
last_enriched - numeric(15,5)
first_enriched - numeric(15,5)
cz_account_id - character varying(512)
observation_count - int
enrich_count - int

- [ ] Existing GitHub Issue? Please provide #: 

### Acceptance checklist

- [ ] Feedback addressed
- [ ] Tests added or modified appropriately
- [ ] All tests passed

### CODE REVIEW

File: reactor/aws/cloudtrail.py [hunk 1]
```diff
@@ -301,8 +301,14 @@ def normalize(self, environments, event_owner_id=None):
...
-            'environment_id': environment['environment_id'],
-            'environment_type': 'AWS',
+            'environment_id': environment['environment_id'],  # obsolete name
+            'cz_account_id': environment['environment_id'],  # new name for this ^^^
+            'environment_type': 'AWS',  # obsolete name
+            'cloud_account_type': 'AWS',  # new name for this ^^^
+            'service_type': None,  # future use
+            'resource_type': None,  # future use
+            'event_service_host': raw_event.get('eventSource'),
+            'event_flags': 0,  # future use
```
[the remark this answers is not in the document]
Comment (killthrush) [reply reactor/aws/cloudtrail.py]: You should be able to now create these columns in Core and update the loaders to read these items even if the values are not yet available

File: reactor/features/resources/enrich.py [hunk 2]
```diff
@@ -171,7 +171,7 @@ def enrich_resource_observations(resource_observations, namespace, reactor_accou
         unenriched_resource['resource_type'] = czrn.extract_resource_type(resource_id)
         unenriched_resource['cloud_local_id'] = czrn.extract_cloud_local_id(resource_id)
         unenriched_resource['resource_name'] = czrn.extract_cloud_local_id(resource_id)
-        unenriched_resource['cloud_account_id'] = account_context['connected_account_id']
+        unenriched_resource['cloud_account_id'] = czrn.extract_account_id(resource_id)
```
Comment (killthrush) [inline reactor/features/resources/enrich.py]: Fix unrelated bug with enrich - the account ID being stored with the resource was the cloud_account_id instead of the account ID loaded from the resource CZRN.  They were swapped

File: reactor/features/resources/reconcile.py [hunk 3]
```diff
@@ -88,9 +88,12 @@ def _initialize_incoming_resource(enriched_resource):
...
     new_resource['cz_metadata'] = {}
-    new_resource['last_seen'] = get_timestamp()
-    if 'first_seen' not in new_resource:
-        new_resource['first_seen'] = new_resource['last_seen']
+    # new_resource['last_observed'] = get_timestamp()
+    if 'first_observed' not in new_resource:
+        new_resource['first_observed'] = new_resource['last_observed']
+    new_resource['last_enriched'] = get_timestamp()
+    if 'first_enriched' not in new_resource:
+        new_resource['first_enriched'] = new_resource['last_enriched']
```
Comment (killthrush) [inline reactor/features/resources/reconcile.py]: No longer using `first_seen` and `last_seen` - these are now `first_observed` and `last_observed`.  We now also have timestamps for enrich.

File: reactor/features/resources/enrich.py [hunk 4]
```diff
@@ -182,12 +182,12 @@ def _run_enrichment_plugin(account_context, enrich_plugin, resource_stream_event
...
-        # Detect if observed resource exists outside of the currently connected account
+        # Detect if observed resource exists outside of the currently connected cloud account
         # We are unable at this time to enrich such things
-        resource_cloud_account_id = czrn.extract_account_id(resource_stream_event['res_id'])
-        if is_external_resource(resource_id, resource_stream_event['cloud_account_id']):
+        cloud_account_id = account_context['connected_account_id']
+        if cloud_account_id != resource_stream_event['cloud_account_id']:
             data = {"message": "Observed resource exists outside account",
-                    "true_cloud_account_id": resource_cloud_account_id}
+                    "true_cloud_account_id": resource_stream_event['cloud_account_id']}
```
Comment (killthrush) [inline reactor/features/resources/enrich.py]: This bug is unrelated to the ticket - the account IDs were swapped so we were not logging the correct account ID if the resource was external

File: reactor/aws/cloudtrail.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (imcgaunn) [reply reactor/aws/cloudtrail.py]: instead of `new name for this ` it would be better to write `new name for environment id`

File: reactor/features/resources/reconcile.py [hunk 5]
```diff
@@ -141,13 +143,13 @@ def _reconcile_resource(existing_resource, enriched_resource):
...
     """
-    curr_timestamp = enriched_resource['last_seen']
+    curr_timestamp = enriched_resource['last_enriched']
     new_resource = copy.deepcopy(enriched_resource)
 
     # Handle enrich and observation counts
     new_resource['count'] += existing_resource['count']  # TODO: remove this when attribute is no longer needed
     new_resource['enrich_count'] += existing_resource['enrich_count']
-    new_resource['observation_count'] += existing_resource.get('observation_count', 0)
+    new_resource['observation_count'] += existing_resource['observation_count']
```
Comment (imcgaunn) [inline reactor/features/resources/reconcile.py]: this is probably safe, or not much more dangerous than defaulting to 0. an observation count of 0 is a death sentence 
for a resource anyway.
